hicommonwealth/commonwealth
# add new counts columns and modified routes queries

https://github.com/hicommonwealth/commonwealth/wiki/Pre-Aggregated-Metrics

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/4201

## Description of Changes
- New columns migration script
- Compute current count to backfill values
- Modified queries based on new column
-  Add hooks to keep counts upto date


### Support for Scheduled job:
```
yarn --cwd packages/commonwealth recompute-counts
```

## Impact
- Speed up route queries - viewUserActivity, globalUserActivity, getThreads, bulkThreads, activeThreads

## Test Plan
- CA (click around) tested on local and frack:
- Load testing to measure performance impact

## Steps for Testing
- login
- go to any community, select thread
   - add/remove reaction to new or existing Thread(s)
   - add/remove new comment to Thread(s)
   - add/remove reaction to comment(s)
   - check all views(topic view, overview) wherever we show comment & reaction count if they are all updated
   - refresh window again - to verify changes are not done on client side only, and are successfully done on backend
- go to user dashboard
   - verify global & user activity feed to show new thread & new comment activity
   - verify new activity in feed.

## Deployment Plan
1. Heroku scheduler new job - to recompute 
- Click add job - choose 03:00AM <timezone> everyday
- Add this job
```
# instructions to run from root folder
yarn --cwd packages/commonwealth recompute-counts
```

## Other Considerations
- we verified job on heroku scheduler on frick - this statement will work
```
yarn --cwd packages/commonwealth recompute-counts
```
